pytorch_lightning/trainer/trainer.py

```diff
@@ -876,6 +867,21 @@ def _run(self, model: 'pl.LightningModule') -> Optional[Union[_EVALUATE_OUTPUT,
         # plugin will setup fitting (e.g. ddp will launch child processes)
         self._pre_dispatch()
 
+        # restore modules after setup
```

Comment:
Is this part of this PR?

Reply:
Was trying it out, but breaks the tests so will be reverting once we get around to merging this!